turbo/jsonrpc/receipts/bor_receipts_generator.go

```diff
@@ -98,17 +86,17 @@ func applyBorTransaction(msgs []*types.Message, evm *vm.EVM, gp *core.GasPool, i
 		lastReceipt = blockReceipts[numReceipts-1]
 	}
 
-	receiptLogs := ibs.Logs()
+	receiptLogs := ibs.GetLogs(0, bortypes.ComputeBorTxHash(block.NumberU64(), block.Hash()), block.NumberU64(), block.Hash())
```

Comment:
are you sure `txIndex = 0` is right for BorTransaction?

Reply:
In this case yes, we create a new ibs just to run the state sync events so it will always be at index 0.

Reply:
@shohamc1 aren't state sync txn's executed at the end of a block - e.g. if a block has 2 transactions (txn1 and txn2), you need to execute those 2 transactions first to update the state of accounts, then after that execute the state sync event txn last? so the pre-state of a bor state sync txn needs to be the post-state after txn2 is executed

Reply:
in essence my question is, is it correct to call the below with txIndex=0 for state sync txns? (taken from `BorGenerator.GenerateBorReceipt`):
```
ibs, blockContext, _, _, _, err := transactions.ComputeBlockContext(ctx, g.engine, block.HeaderNoCopy(), chainConfig, g.blockReader, txNumsReader, tx, 0)
```

Reply:
I guess it boils down to what is the behaviour of 
```
reader, err := rpchelper.CreateHistoryStateReader(dbtx, txNumsReader, header.Number.Uint64(), txIndex, cfg.ChainName)
```

If we call it for Block B1 with txIndex=0 does it give us the state the world as of before execution any txns from B1? If so then shouldn't we call it with txIndex=len(B1.Transactions())-1 so that the correct state is available for when we execute the state sync txn at the end of the block?

Reply:
Ah this is a different matter from Alex's inquiry. Looking into it, it seems this was an oversight and you are correct. I have opened a [PR](https://github.com/erigontech/erigon/pull/12408) to fix this.